File: 1975-maximum-matrix-sum/1975-maximum-matrix-sum.java
class Solution {
    public long maxMatrixSum(int[][] matrix) {
        long totalSum = 0;
        int minValue = Integer.MAX_VALUE;
        int negativeCount = 0;

        for (int[] row : matrix) {
            for (int x : row) {
                totalSum += Math.abs(x);
                minValue = Math.min(minValue, Math.abs(x));
                if (x < 0) {
                    negativeCount++;
                }
            }
        }

        return totalSum - (negativeCount % 2 != 0 ? 2L * minValue : 0);
    }
}

File: 1975-maximum-matrix-sum/1975-maximum-matrix-sum.js
var maxMatrixSum = function(matrix) {
    let totalSum = 0;
    let minValue = Infinity;
    let negativeCount = 0;

    for (const row of matrix) {
        for (const x of row) {
            totalSum += Math.abs(x);
            minValue = Math.min(minValue, Math.abs(x));
            if (x < 0) {
                negativeCount++;
            }
        }
    }

    return totalSum - (negativeCount % 2 !== 0 ? 2 * minValue : 0);
};
